src/bun.js/bindings/PCRE2RegExp.cpp

```diff
@@ -0,0 +1,790 @@
+#include "PCRE2RegExp.h"
+
+#include "ZigGlobalObject.h"
+#define PCRE2_CODE_UNIT_WIDTH 16
+#include "pcre2.h"
+
+using namespace JSC;
+using namespace WebCore;
+
+#include "WebCoreJSClientData.h"
+
+extern "C" EncodedJSValue jsFunctionGetPCRE2RegExpConstructor(JSGlobalObject * lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName)
+{
+    auto& vm = lexicalGlobalObject->vm();
+    Zig::GlobalObject *globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
+    return JSValue::encode(globalObject->PCRE2RegExpConstructor());
+}
+
+namespace Zig {
+
+static WTF::String to16Bit(ASCIILiteral str)
+{
+    return WTF::String::make16BitFrom8BitSource(str.characters8(), str.length());
+}
+
+static  WTF::String to16Bit(JSC::JSString* str, JSC::JSGlobalObject *globalObject) {
+    if (!str->is8Bit() || str->length() == 0) {
+        return str->value(globalObject);
+    }
+
+    auto value = str->value(globalObject);
+    return WTF::String::make16BitFrom8BitSource(value.characters8(), value.length());
+}
+
+
+static  WTF::String to16Bit(JSValue jsValue, JSC::JSGlobalObject *globalObject, ASCIILiteral defaultValue) {
+    if (!jsValue || jsValue.isUndefinedOrNull()) {
+        return  to16Bit(defaultValue);
+    }
+
+    auto *jsString = jsValue.toString(globalObject);
+    if (jsString->length() == 0) {
+        return to16Bit(defaultValue);
+    }
+
+    return to16Bit(jsString, globalObject);
+}
+
+static inline bool is16BitLineTerminator(UChar c)
+{
+    return c == '\r' || c == '\n' || (c & ~1) == 0x2028;
```

Comment:
There are technically about 10 more of these but I assume this was copied from RegExpPrototype.cpp in WebKit so it's probably fine